### PR TITLE
Avoid allocations for binding values

### PR DESCRIPTION
**Pull request by charliermarsh**
This achieves a 1-1.5% performance improvement on the benchmark (shrug).

### CODE REVIEW

File: src/ast/helpers.rs [hunk 1]
```diff
@@ -225,6 +225,15 @@ pub fn is_super_call_with_arguments(func: &Expr, args: &[Expr]) -> bool {
     }
 }
 
+/// Format the module name for a relative import.
+pub fn format_relative_import(level: &Option<usize>, module: &Option<String>) -> String {
+    format!(
+        "{}{}",
+        ".".repeat(level.unwrap_or_default()),
+        module.clone().unwrap_or_default()
```
Comment (Stranger6667) [inline src/ast/helpers.rs:233]: I think that this clone could be avoided as well. E.g. if the string is constructed manually with a series of “push_str” and “push” calls

File: src/ast/helpers.rs [hunk 2]
```diff
@@ -225,6 +225,15 @@ pub fn is_super_call_with_arguments(func: &Expr, args: &[Expr]) -> bool {
     }
 }
 
+/// Format the module name for a relative import.
+pub fn format_relative_import(level: &Option<usize>, module: &Option<String>) -> String {
```
Comment (messense) [inline src/ast/helpers.rs:229]: From Rust API point of view, I think it's kinda odd to pass around `&Option<T>`, `Option<&T>` is much more natural. You can get a `Option<&T>` via [`Option::as_ref()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref).

And to get a `Option<&str>` from a `&Option<String>` you can use [Option::as_deref()](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref).

File: src/ast/helpers.rs (hunk 1 above)
Comment (Stranger6667) [reply src/ast/helpers.rs:233]: To compare these approaches I benched the following:

```rust
pub fn original(level: &Option<usize>, module: &Option<String>) -> String {
    format!(
        "{}{}",
        ".".repeat(level.unwrap_or_default()),
        module.clone().unwrap_or_default()
    )
}

pub fn second(level: &Option<usize>, module: &Option<String>) -> String {
    let mut out = String::new();
    for _ in 0..level.unwrap_or_default() {
        out.push('.');
    }
    if let Some(m) = module {
        out.push_str(m.as_str());
    }
    out
}

pub fn third(level: Option<usize>, module: Option<&str>) -> String {
    let mut out = String::with_capacity(16);
    for _ in 0..level.unwrap_or_default() {
        out.push('.');
    }
    if let Some(m) = module {
        out.push_str(m);
    }
    out
}

fn simple(c: &mut Criterion) {
    let level = black_box(3);
    let module = black_box("foobarbaz".to_string());
    let module_str = black_box("foobarbaz");
    c.bench_function("original", |b| b.iter(|| original(&Some(level), &Some(module.clone()))));
    c.bench_function("second", |b| b.iter(|| second(&Some(level), &Some(module.clone()))));
    c.bench_function("third", |b| b.iter(|| {
        // All benches make a String clone
        let _ = module.clone();
        third(Some(level), Some(module_str))
    }));
}
```


And here are the results:

```
original                time:   [97.946 ns 98.351 ns 98.802 ns]                     
                        change: [-0.6825% -0.2181% +0.3228%] (p = 0.45 > 0.05)
                        No change in performance detected.
Found 13 outliers among 100 measurements (13.00%)
  1 (1.00%) low mild
  8 (8.00%) high mild
  4 (4.00%) high severe

second                  time:   [50.236 ns 50.383 ns 50.532 ns]                    
                        change: [+1.7089% +2.0963% +2.5148%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  2 (2.00%) low mild
  2 (2.00%) high mild

third                   time:   [29.787 ns 29.853 ns 29.922 ns]                   
                        change: [-0.7170% -0.2109% +0.3007%] (p = 0.42 > 0.05)
                        No change in performance detected.
Found 6 outliers among 100 measurements (6.00%)
  2 (2.00%) high mild
  4 (4.00%) high severe
```

It is absolutely not exhaustive, but I think it gives some intuition on what takes how much. The string capacity is somewhat arbitrary.

File: src/ast/helpers.rs (hunk 2 above)
Comment (charliermarsh) [reply src/ast/helpers.rs:229]: Oh, thank you! That's really helpful feedback. I'm gonna fix this everywhere.

File: src/ast/helpers.rs (hunk 1 above)
Comment (charliermarsh) [reply src/ast/helpers.rs:233]: This is great, thanks! Always appreciate a benchmark :)

**Pull request merged**